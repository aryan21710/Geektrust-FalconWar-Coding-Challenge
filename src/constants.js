export const UNITSANDTRAVELTIMEDATA = Array(4).fill({
	name: '',
	botImageName: '',
	distance: '',
	speed: '',
	travelTime: 0,
	totalUnits: {
		original: -1,
		current: -1,
	},
	error: false,
	planetIndexArr: [],
	vehicleIndexArr: [],
});

export const DATATOANIMSELECTEDPLANET = Array(6)
	.fill('')
	.map(() => ({
		isAnimated: false,
		imgname: '',
		index: -1,
		planetname: '',
		distance: '',
	}));
