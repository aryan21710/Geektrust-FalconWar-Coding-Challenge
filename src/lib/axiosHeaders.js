export const defaultAxiosHeader = {
	mode: 'cors',
	cache: 'no-cache',
	credentials: 'same-origin',
	redirect: 'follow',
	referrerPolicy: 'no-referrer',
};

export const postDataAxiosHeader = {
	Accept: 'application/json',
	'Content-Type': 'application/json',
};
