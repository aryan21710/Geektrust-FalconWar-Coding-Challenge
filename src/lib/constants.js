import Planet1 from '../assets/images/1.png';
import Planet2 from '../assets/images/2.png';
import Planet3 from '../assets/images/3.png';
import Planet4 from '../assets/images/4.png';
import Planet5 from '../assets/images/5.png';
import Planet6 from '../assets/images/6.png';
import KingShan from '../assets/images/modiji.png';
import ALFALCONE from '../assets/images/xi jhi.png';
import Minijet from '../assets/images/minijet.png';
import SpaceRocket from '../assets/images/Spaceship1.png';
import SpacePod from '../assets/images/spaceship2.png';
import SpaceShip from '../assets/images/Spaceship3.png';
import SpaceShuttle from '../assets/images/spaceship4.png';

export const PlanetImageArr = [Planet1, Planet2, Planet3, Planet4, Planet5, Planet6];

export const ImageBadges = {
	KingShan,
	ALFALCONE,
};

export const MinijetImage = {
	Minijet,
};

export const SpaceBotImgArr = [SpacePod, SpaceRocket, SpaceShuttle, SpaceShip];
export const UNITSANDTRAVELTIMEDATA = Array(4).fill({
	name: '',
	botImageName: '',
	distance: '',
	speed: '',
	travelTime: 0,
	totalUnits: {
		original: -1,
		current: -1,
	},
	error: false,
	planetIndexArr: [],
	vehicleIndexArr: [],
});

export const DEFAULTDATATOANIMSELECTEDPLANET = Array(6)
	.fill('')
	.map(() => ({
		isAnimated: false,
		imgname: '',
		index: -1,
		planetname: '',
		distance: '',
	}));

export const DEFAULTPLANETANDBOTSDATA = {
	planetname: '',
	planetindex: -1,
	imgname: '',
	distance: 0,
	animPlanetCnt: 0,
	planetNameArr: [],
};


export const MAPPINGBETNPLANETANDBOTSDATA = [
	{
		planetname: 'DONLON',
		imgname: Planet1,
		distance: 100,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 50,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 25,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 20,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 10,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
	{
		planetname: 'JEBING',
		imgname: Planet3,
		distance: 300,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 0,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 75,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 60,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 30,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
	{
		planetname: 'LERBIN',
		imgname: Planet5,
		distance: 500,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 0,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 0,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 0,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 50,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
	{
		planetname: 'PINGASOR',
		imgname: Planet6,
		distance: 600,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 0,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 0,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 0,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 60,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
	{
		planetname: 'SAPIR',
		imgname: Planet2,
		distance: 400,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 0,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 0,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 80,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 40,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
	{
		planetname: 'ENCHAI',
		imgname: Planet4,
		distance: 200,
		planetToVehMapping: [],
		vehicleDataArray: [
			{
				name: 'SPACEPOD',
				botImageName: '',
				travelTime: 100,
				totalUnits: 2,
				maxUnits: 2,

			},
			{
				name: 'SPACEROCKET',
				botImageName: '',
				travelTime: 50,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHUTTLE',
				botImageName: '',
				travelTime: 40,
				totalUnits: 1,
				maxUnits: 1,

			},
			{
				name: 'SPACESHIP',
				botImageName: '',
				travelTime: 20,
				totalUnits: 2,
				maxUnits: 2,

			},
		],
	},
];
