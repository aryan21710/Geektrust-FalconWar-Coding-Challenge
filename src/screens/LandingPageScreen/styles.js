import styled from 'styled-components';
import { Heading } from '../../components/common/styles';

export const Wrapper = styled.div`
	width: 100vw;
	height: 100vh;
	margin: 0;
	padding: 0;
	overflow: hidden;
	display: flex;
	justify-content: center;
	opacity: 1;
	z-index: 2;
	flex-direction: ${(props) => props.flexDirection || 'row'};
	@media (max-width: 768px) {
		flex-direction: column;
		height: 80vh;
	}
`;

export const BadgeWrapper = styled.div`
	display: flex;
	justify-content: center;
	align-items: ${(props) => props.alignItems || 'center'};
	justify-content: ${(props) => props.justifyContent || 'center'};
	flex-direction: column;
	height: ${(props) => props.height || '75vh'};
	width: ${(props) => props.width || '25vw'};
	flex: ${(props) => props.flex || '1'};
	flex-wrap: ${(props) => props.flexWrap};
	z-index: 2;
	@media (max-width: 768px) {
		width: 100vw;
		height: 30vh;
	}
`;

export const ImageWrapper = styled.img`
	border-radius: ${(props) => props.borderRad || '50%'};
	margin-bottom: 3vh;
	height: ${(props) => props.height || '30vh'};
	width: ${(props) => props.height || '30vh'};
	object-fit: ${(props) => props.objectFit || 'cover'};
	transform: rotate(${(props) => props.rotateBy || 'none'});
	@media (max-width: 768px) {
		height: 20vh;
		width: 20vh;
	}
`;

export const ButtonWrapper = styled.div`
	width: 100vw;
	height: 20vh;
	position: absolute;
	bottom: 5vh;
	display: flex;
	justify-content: space-around;
	align-items: center;
	flex-direction: column;
`;

export const BigHeading = styled(Heading)`
	font-size: 1.5rem;
	color: '#FAD107';
	text-align: center;
	font-family: ${(props) => props.fontFamily || 'Nasalisation'};
	z-index: 1000;
	font-weight: 100;
	margin: ${(props) => props.margin || '2.5px auto'};
	@media (max-width: 768px) {
		font-size: 1rem;
	}
`;

export const SmallHeading = styled(BigHeading)`
	font-size: 1.2rem;
	color: 'white';
	font-family: 'Avenir';
`;
