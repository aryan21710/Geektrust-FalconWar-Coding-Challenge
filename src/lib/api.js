import axios from 'axios';
import { defaultAxiosHeader, postDataAxiosHeader } from './axiosHeaders';

export const makePostRequestToBackend = async (url, setApiError, body = undefined) => {
	try {
		return await axios.post(url, body, {
			headers: postDataAxiosHeader,
		});
	} catch (err) {
		err?.response &&
			setApiError({
				status: err.response.status,
				data: err.response.data,
				message: err.message,
				response: err.response,
				url: err.response.config.url,
			});
	}
};

export const makeGetRequestToBackend = async (url, setApiError) => {
	try {
		return await axios.get(url, {
			defaultAxiosHeader,
		});
	} catch (err) {
		err?.response &&
			setApiError({
				status: err.response.status,
				data: err.response.data,
				message: err.message,
				response: err.response,
				url: err.response.config.url,
			});
	}
};
