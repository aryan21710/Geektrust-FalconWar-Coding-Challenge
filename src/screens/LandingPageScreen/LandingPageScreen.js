import React, { useContext } from 'react';
import {
	Wrapper,
	BadgeWrapper,
	ImageWrapper,
	ButtonWrapper,
	SmallHeading,
	BigHeading
} from './styles.js';
import { useFetchDataFromBackend } from '../services';
import { ImageBadges } from '../../lib/constants';
import { PlanetDetailsContext } from '../../context/appContext';
import { CustomButton } from '../../sharedComponents/CustomButton';
import uuid from 'react-uuid';

const LandingPageScreen = () => {
	const { planetCfg, setPlanetCfg, apiError, setApiError } = useContext(PlanetDetailsContext);
	useFetchDataFromBackend(planetCfg, setPlanetCfg, apiError, setApiError);

	return (
		<React.Fragment>
			<Wrapper >
				{Object.keys(ImageBadges).map((image) => (
					<BadgeWrapper  key={uuid()}>
						<ImageWrapper id="images" src={ImageBadges[image]} />
						<BigHeading>
							{image}
						</BigHeading>
					</BadgeWrapper>
				))}

				<ButtonWrapper>
					<SmallHeading id="heading">
						Queen Al Falcone is now in hiding. But if King Shan can find her before the years are up, she
						will be exiled for another 15 yearsâ€¦
					</SmallHeading>
					<CustomButton
						disabled={Object.keys(apiError).length === 0 ? false : true}
						redirectPath="/selectplanets"
						leftpos="0vh"
						opacity={Object.keys(apiError).length === 0 ?  1 : 0.6}
						TextForButton="Lets Find Falcone"
					/>
				</ButtonWrapper>
			</Wrapper>
		</React.Fragment>
	);
};

export default LandingPageScreen;
