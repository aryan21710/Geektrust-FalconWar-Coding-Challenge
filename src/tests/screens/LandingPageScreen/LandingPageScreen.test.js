import React from 'react';
import LandingPageScreen from '../../../screens/LandingPageScreen/LandingPageScreen';
import { shallow, mount, render } from '../../../../enzyme';
import toJSON from 'enzyme-to-json';
import { PlanetDetailsContext } from '../../../context/appContext';
import { ImageBadges } from '../../../lib/constants';

describe('TESTING LandingPageScreen COMPONENT', () => {
	test('SNAPSHOT', () => {
		const TestComponent = () => {
			return (
				<PlanetDetailsContext.Provider value={PlanetDetailsContext}>
					<LandingPageScreen />
				</PlanetDetailsContext.Provider>
			);
		};

		const wrapper = shallow(<TestComponent />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	test('renders text properly', () => {
		const TestComponent = () => {
			return (
				<PlanetDetailsContext.Provider
					value={{ planetCfg: {}, setPlanetCfg: {}, apiError: {}, setApiError: {} }}
				>
					<LandingPageScreen />
				</PlanetDetailsContext.Provider>
			);
		};

		const wrapper = mount(<TestComponent />);
		const wrapperB = wrapper.find('img#images');
		console.log('image:-', wrapper.debug());
		expect(wrapper.find('img#images')).toHaveLength(Object.keys(ImageBadges).length);
        expect(wrapper.find(LandingPageScreen).exists()).toBeTruthy();
        expect(wrapper.find('h1#heading').exists()).toBeTruthy();
        expect(wrapper.find('h1#heading').text()).toMatch(/Queen Al Falcone/);


	});
});
