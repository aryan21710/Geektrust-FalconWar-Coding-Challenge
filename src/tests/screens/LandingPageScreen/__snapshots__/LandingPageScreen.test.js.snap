// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING LandingPageScreen COMPONENT SNAPSHOT 1`] = `
<ContextProvider
  value={
    Object {
      "$$typeof": Symbol(react.context),
      "Consumer": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": [Circular],
      },
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "_threadCount": 0,
    }
  }
>
  <LandingPageScreen />
</ContextProvider>
`;
